blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineImpl.java

```diff
@@ -358,6 +370,160 @@ public Resource resolve(Item context, Reachable parent, Item target) {
 
     }
 
+    @Extension
+    public static class PipelineCreatorFactoryImpl extends BluePipelineCreatorFactory{
+        @Override
+        public BluePipeline create(Map<String, Object> request, Reachable parent) throws IOException {
+            ScmOrganizationPipelineRequest scmOrganizationPipelineRequest = ScmOrganizationPipelineRequest.getInstance(request);
+            if(scmOrganizationPipelineRequest == null){
+                return null;
+            }
+            ScmProvider scmProvider=null;
+            for(ScmProviderFactory f:ScmProviderFactory.all()){
+                scmProvider = f.getScmProvider(scmOrganizationPipelineRequest.scmProviderId);
+                if(scmProvider == null){
+                    continue;
+                }
+                break;
+            }
+            if(scmProvider == null){
+                throw new ServiceException.BadRequestExpception("Unkown scmProviderId: "+ scmOrganizationPipelineRequest.scmProviderId);
+            }
+
+//            CredentialsMatcher matcher = CredentialsMatchers.withId(scmOrganizationPipelineRequest.credentialId);
+//            List<StandardUsernamePasswordCredentials> credentials=CredentialsProvider.lookupCredentials(StandardUsernamePasswordCredentials.class);
+//            StandardUsernamePasswordCredentials cred = CredentialsMatchers.firstOrNull(credentials, matcher);
+//            if(cred == null){
+//                throw new ServiceException.BadRequestExpception("Credential with id: "+ scmOrganizationPipelineRequest.credentialId+" not found");
+//            }
+
+            return scmProvider.createProjects(scmOrganizationPipelineRequest, Jenkins.getInstance(), parent.getLink());
+        }
+    }
+
+    @Extension
+    public static class GithubScmProviderFactory extends ScmProviderFactory implements ScmProvider{
```

Comment:
I guess this should be in its own class to allow github to not be coupled to the generic multibranchpipeline class